doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+customizations to the settings. These two files would then be composed together
+when the app is launched, so that the runtime settings are the union of both the
+defaults and whatever modifications the user has chosen. This will enable the
+app to always use a default schema that it knows will be valid, and minimize the
+settings that the user needs to customize.
+
+Should the settings schema ever change, the defaults file will change, without
+needing to re-write the user's settings file.
+
+
+## Inspiration
+
+Largely inspired by the settings model that both VS Code (and Sublime Text) use.
+
+### Goal: Minimize Writing to `profiles.json`
+
+We want to write the user settings file, `profiles.json`, as little as possible.
+Each time we write the file, there's the possiblity that we've re-ordered the
+keys, as `jsoncpp` provides no ordering guarantee of the keys. This isn't great,
+as each write of the file will randomly re-order the file.
+
+One of our overarching goals with this change should be to re-write the user
+settings file as little as possible.
+
+## Solution Design
+
+The settings are now composed from two files: a "Default" settings file, and a
+"User" settings file.
+
+### Default Settings
+
+We'll have a static version of the "Default" file hardcoded within the
+application. We'll write the hardcoded version of this file out to an actual
+file on disk, `defaults.json`, however, that file will not actually be used. The
+defaults file will contain some sort of disclaimer like:
+
+```json
+// This is an auto-generated file. Place any modifications to your settings in "profiles.json"
+```
+
+This disclaimer will help identify that the file shouldn't be modified.
+
+The `defaults.json` file will be written out whenever we detect that it doesn't exist.
+
+**TODO**: We also need to write it out if it changes, though ideally without
+actually loading the file at first load. Perhaps we could start another
+thread/`fire_and_forget` once we're loaded to re-scan the file and re-write it
+if it has changed.
+
+Because the `defaults.json` file is hardcoded within our application, we can use
+its text directly, without loading the file from disk. This should help save
+some startup time, as we'll only need to load the user settings from disk.
+
+When we make changes to the default settings, or we make changes to the settings
+schema, we should make sure that we update the hardcoded `defaults.json` with
+the new values. That way, the `defaults.json` file will always have the complete
+set of setings in it.
+
+
+### Layering settings
+
+When we load the settings, we'll do it in two stages. First, we'll deseriale the
+default settings that we've hardcoded. Then, we'll intelligently layer the
+user's setting upon those we've already loaded. Some objects, like the default
+profiles, we'll need to make sure to load from the user settings into the
+existing objects we created from the default settings.

```

Comment:
Same for color schemes.

Reply:
I thought I kinda talked about that in the subsequent bullet points...